pandas/core/interchange/column.py

```diff
@@ -302,15 +326,27 @@ def _get_data_buffer(
             DtypeKind.FLOAT,
             DtypeKind.BOOL,
         ):
+            dtype = self.dtype
             arr = self._col.array
+            if isinstance(self._col.dtype, ArrowDtype):
+                buffer = PandasBufferPyarrow(
+                    arr._pa_array,  # type: ignore[attr-defined]
+                    is_validity=False,
+                    allow_copy=self._allow_copy,
+                )
+                if self.dtype[0] == DtypeKind.BOOL:
```

Comment:
Can you clarify what the purpose of this is? Can the dtype not simply be forwarded along?

Reply:
indeed, it was unnecessary, thanks!